torch/_inductor/fx_passes/group_batch_fusion.py

```diff
@@ -740,46 +752,105 @@ def __init__(self, **kwargs):
         super().__init__(aten.mul.Tensor, **kwargs)
 
 
+class _OrderedSet:
```

Comment:
In Python, dicts are now ordered, so you should instead use a Dict[Key, None] for this instead of rolling out your own ordered set.

Reply:
This is just an encapsulation of Dict[Key, None] (via OrderedDict). I would think Isn't it more clear to a reader that the code isn't relying on the dict value by eliminating it?

Reply:
We have quite a lot of this pattern in the codebase. Not the best regex but you can browse: https://fburl.com/biggrep/cp0sy1sg

If we want to proceed with the OrderedSet wrapper, it might be good to unify these, although someone more familiar with the codebase should pitch in here. @ezyang ?

Reply:
Yea I think OrderedSet would be an improvement throughout the codebase (especially in compiler). let's leave that for another PR though.